src/testSession.ts

```diff
@@ -235,55 +245,83 @@ export class TestSession {
+              this.setup.push({ result: { username: org } });
+              continue;
+            }
           }
-        }
 
-        // Add the json flag if it looks like an sfdx command so we can return
-        // parsed json in the command return.
-        if (cmd.split(' ')[0].includes('sfdx') && !cmd.includes('--json')) {
-          cmd += ' --json';
-        }
+          // Add the json flag if it looks like an sfdx command so we can return
+          // parsed json in the command return.
+          if (cmd.split(' ')[0].includes('sfdx') && !cmd.includes('--json')) {
+            cmd += ' --json';
+          }
 
-        const rv = shell.exec(cmd, { silent: true });
-        if (rv.code !== 0) {
-          const io = cmd.includes('--json') ? rv.stdout : rv.stderr;
-          throw Error(`Setup command ${cmd} failed due to: ${io}`);
-        }
-        dbug(`Output for setup cmd ${cmd} is:\n${rv.stdout}`);
-
-        // Automatically parse json results
-        if (cmd.includes('--json')) {
-          try {
-            const jsonOutput = parseJson(rv.stdout);
-            // keep track of all org creates
-            if (cmd.includes('org:create')) {
-              const username = getString(jsonOutput, 'result.username');
-              if (username) {
-                dbug(`Saving org username: ${username} from ${cmd}`);
-                this.orgs.push(username);
+          const rv = shell.exec(cmd, { silent: true });
+          // const rv = this.execWithRetries(cmd);
+          if (rv.code !== 0) {
+            const io = cmd.includes('--json') ? rv.stdout : rv.stderr;
+            throw Error(`Setup command ${cmd} failed due to: ${io}`);
+          }
+          dbug(`Output for setup cmd ${cmd} is:\n${rv.stdout}`);
+
+          // Automatically parse json results
+          if (cmd.includes('--json')) {
+            try {
+              const jsonOutput = parseJson(rv.stdout);
+              // keep track of all org creates
+              if (cmd.includes('org:create')) {
+                const username = getString(jsonOutput, 'result.username');
+                if (username) {
+                  dbug(`Saving org username: ${username} from ${cmd}`);
+                  this.orgs.push(username);
+                }
               }
+              this.setup.push(jsonOutput);
+            } catch (err: unknown) {
+              dbug(`Failed command output JSON parsing due to:\n${(err as Error).message}`);
+              this.setup.push(rv);
             }
-            this.setup.push(jsonOutput);
-          } catch (err: unknown) {
-            dbug(`Failed command output JSON parsing due to:\n${(err as Error).message}`);
+          } else {
             this.setup.push(rv);
           }
-        } else {
-          this.setup.push(rv);
         }
       }
+    };
+
+    let attempts = 0;
+    let completed = false;
+    const timeout = new Duration(env.getNumber('TESTKIT_SETUP_RETRIES_TIMEOUT') ?? 5000, Duration.Unit.MILLISECONDS);
```

Comment:
nit: I think `getNumber` has a 2nd arg that is the default if undefined.

Reply:
That's what I originally had but I switched to this because typescript thinks that the default value is `undefined`. Using `??` mitigates the need to redundantly state `as number`